src/restore.ts

```diff
@@ -35,22 +35,46 @@ async function run(): Promise<void> {
             restoreKeys
         );
 
+        //Check if user wants to save cache despite of failure in any previous job
+        const saveCache = core.getInput(Inputs.SaveOnAnyFailure).toLowerCase();
+        if (saveCache == "true") {
+            core.debug(
+                `Exporting environment variable ${Variables.SaveCacheOnAnyFailure}`
+            );
+            core.exportVariable(Variables.SaveCacheOnAnyFailure, saveCache);
+            core.info(
+                `Input Variable ${Variables.SaveCacheOnAnyFailure} is set to true, the cache will be saved despite of any failure in the build.`
+            );
+        }
+
         if (!cacheKey) {
+            if (core.getInput(Inputs.FailOnCacheMiss).toLowerCase() == "true") {
+                throw new Error(
+                    `Cache with the given input key ${primaryKey} is not found, hence exiting the workflow as the fail-on-cache-miss requirement is not met.`
+                );
+            }
             core.info(
                 `Cache not found for input keys: ${[
                     primaryKey,
                     ...restoreKeys
                 ].join(", ")}`
             );
-
             return;
         }
-
         // Store the matched cache key
         utils.setCacheState(cacheKey);
```

Comment:
Wondering if we should not set this state (and the primary key state in L25) if this flow is for `restore` action. Since there is no post step, it does not make sense to set these states.

Reply:
One thing I am thinking is:
- If an output, set it irrespective of their granular actions or `cache` action
- If a state, set it only for `cache` action

Reply:
Which means the wrapper classes would take care of setting as an output or a state. Is that right?

Reply:
Yes @kotewar

Reply:
Even the non common inputs handling can be part of the wrapper.